keps/sig-node/753-keystone-containers/README.md

```diff
@@ -0,0 +1,896 @@
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+## Motivation
+
+There are lots of [user requests](https://https://github.com/kubernetes/kubernetes/issues/25908) for better support of sidecar containers in Jobs but this issue mostly gets lost in a larger attempt to address the sidecar issue as a whole.
+
+Consider a K8s job with two containers one that does the actual work and another one is for monitoring/logging, the keystone container will complete the work and exit, the sidecar container will keep running and the job will never run to completion. This is problematic for any external application/tool depending on the job waiting to gets completed. Also, the sidecars unnecessarily occupy the system resources.
+
+The only workaround is that the sidecar container needs some kind of signaling from the app container when it exits or the sidecar monitors the app container status. This is not the ideal workaround as it requires containers to be platform aware.
+
+Another motivation for this KEP is to take small steps towards the sidecar containers that will help a lot of users without including complicated logic and changing critical kubelet code paths.
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+### Goals
+- Pod termination by kubelet if application containers have successfully run to completion and exited.
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+### Non-Goals
+- Handling Container startup/shutdown ordering for application and sidecar containers.
+- Pod termination by kubelet if sidecar containers have exited (crash).  There will be no change in pod lifecycle if sidecar container exited.
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
+A container API field (I'm a keystone) can be specified by users to identify each core application containers in the pod.
+
+This identification of application containers is used by kubelet to make a decision on pod termination based on the status of application containers and pod restart policy.
```

Comment:
suggest using "keystone containers"  instead of "application containers" to be consistent with the term defined earlier.